File: 1013-fibonacci-number/fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        cur = 1
        prev = 0
        if n == cur or n == prev : return n
        for i in range(2,n+1) :
            temp = cur + prev
            prev = cur
            cur = temp
        return cur

File: 1013-fibonacci-number/fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        if(n == 0 || n == 1) return n;
        int prev = 0;
        int cur = 1;
        for(int i = 2; i <= n; i++) {
            int temp = prev;
            prev = cur;
            cur = cur + temp;
        }
        return cur;
    }
};